pkgs/by-name/mi/missidentify/package.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, fetchurl
+, fetchpatch
+, autoreconfHook
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "missidentify";
+  version = "1.0";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/project/missidentify/missidentify/missidentify-${finalAttrs.version}/missidentify-${finalAttrs.version}.tar.gz";
+    hash = "sha256-nnIRN8hpKM0IZCe0HUrrJGrxvBYKeBmdU168rlo8op0=";
+  };
+
+  patches = [
+    # define PATH_MAX variable to fix a FTBFS in Hurd.
+    (fetchpatch {
```

Comment:
`fetchpatch` is only needed over `fetchurl` when the patch is dynamically generated from a source control system. This seems to be a static file so `fetchurl` should be fine.

Reply:
@emilytrau how can I recognize if a patch from an URL is dynamically generated or it is a static file?

Reply:
An example of a dynamically generated patch would be [GitHub's generated diff for this PR](https://github.com/NixOS/nixpkgs/pull/289535.patch). A file that's inside a repo would be considered static.

Reply:
Please add the `name` attribute to the `fetchurl` patches as well

Reply:
@emilytrau fixed. Thanks for the explanation.